appservice/src/createAppService/setLocationsTask.ts

```diff
@@ -32,7 +36,15 @@ export async function getWebLocations(context: IAppServiceWizardContext): Promis
         options.sku = <SkuName>context.newPlanSku.tier.replace(/\s/g, '');
     }
 
-    const client: WebSiteManagementClient = await createWebSiteClient(context);
-    const locations = await uiUtils.listAllIterator(client.listGeoRegions(options));
-    return locations.map(l => nonNullProp(l, 'name'));
+    const genericClient = await createGenericClient(context, context);
+    const result: HttpOperationResponse = await genericClient.sendRequest({
+        method: 'GET',
+        pathTemplate: `/providers/Microsoft.Web/geoRegions`,
+        queryParameters: {
+            'api-version': '2020-09-01',
+            ...options,
+        }
+    });
+
+    return (<GeoRegionJsonResponse>result.parsedBody).value.map((l: GeoRegion) => nonNullProp(l, 'name')) as string[];
```

Comment:
This line scares me a bit 😱 I think this accomplishes the same thing with less typecasting:
```ts
    return result.parsedBody?.value.map((l: any) => nonNullProp(l, 'name')) || [];
```

Or maybe we throw an error if `parsedBody` doesn't have a `value` property so we can track it

Reply:
I removed the `as string[]`, it wasn't actually needed. We cast `parsedBody` quite a bit:
https://github.com/microsoft/vscode-azuretools/blob/83d87d1a9a51849cbdf93a7837c02045853c47a7/appservice/src/SiteClient.ts#L316-L323